persistent/Database/Persist/Class/PersistQuery.hs

```diff
@@ -54,6 +57,31 @@ class (PersistCore backend, PersistStoreRead backend) => PersistQueryRead backen
     exists :: (MonadIO m, PersistRecordBackend record backend)
            => [Filter record] -> ReaderT backend m Bool
 
+-- | Backends supporting conditional read operations, which may also support
+-- streaming the results in a memory-constant way.
+--
+-- @since 2.13.2.0
+class (PersistQueryRead backend) => PersistQueryStream backend where
+    -- | Get all records matching the given criterion in the specified order.
+    --
+    -- A version of 'selectSourceRes' which specifically streams the results,
+    -- for SQL backends that support it. Streaming may be slower for small
+    -- query sets, but avoids allocating all the results in memory at once.
+    --
+    -- By default, this behaves identically to 'selectSourceRes'.
+    --
+    -- @since 2.13.2.0
+    selectSourceStream
+        :: (PersistRecordBackend record backend, MonadResource m)
+        => [Filter record]
+        -> [SelectOpt record]
+        -> ConduitM () (Entity record) (ReaderT backend m) ()
+    selectSourceStream filts opts = do
+      srcRes <- lift $ selectSourceRes filts opts
```

Comment:
Eg - an `instance PersistQueryStream MongoContext` would compile just fine, but it would probably fail to work in a reasonable way. That seems bad to me.